app/services/FinancialDetailsService.scala

```diff
@@ -43,9 +43,9 @@ class FinancialDetailsService @Inject()(val financialDetailsConnector: Financial
     financialDetailsConnector.getFinancialDetails(taxYearFrom, taxYearTo, nino)
   }
 
-  def getChargeDueDates(financialDetails: List[FinancialDetailsResponseModel]): Option[Either[(LocalDate, Boolean), Int]] = {
+  def getChargeDueDates(financialDetails: List[FinancialDetailsResponseModel], penaltiesEnabled: Boolean): Option[Either[(LocalDate, Boolean), Int]] = {
```

Comment:
same here, we should not extend FinancialDetailsResponseModel; apply conversion to ChargeItem instead